src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -232,14 +235,63 @@ protected void onStartup() {
         super.onStartup();
 
         initializeReference();
+
         initializeReads();
+
         initializeFeatures();
+
         initializeIntervals(); // Must be initialized last, since intervals currently require a sequence dictionary from another data source
 
+        validateSequenceDictionaries();
+
         checkToolRequirements();
     }
 
     /**
+     * Validate all sequence dictionaries by checking them against each other.
+     */
+    private void validateSequenceDictionaries() {
+        SAMSequenceDictionary refDict = null;
+        if (hasReference()){
+            refDict = reference.getSequenceDictionary();
+        }
+        SAMSequenceDictionary readDict = null;
+        if (hasReads()) {
+            readDict = reads.getSequenceDictionary();
+        }
+        List<SAMSequenceDictionary> featureDictList = new ArrayList<>();
+        if (hasFeatures()){
+            List<VCFHeader> featureHeaders = features.getAllVariantHeaders();
+            for (VCFHeader header : featureHeaders) {
+                SAMSequenceDictionary headerDict = header.getSequenceDictionary();
+                if (headerDict != null) {
+                    featureDictList.add(headerDict);
+                }
+            }
+        }
+        //check reference dict against reads dict
+        if (hasReference() && hasReads()) {
+            SequenceDictionaryUtils.validateDictionaries("reference", refDict, "reads", readDict, true, null);
```

Comment:
Pass in our intervals as the final argument if `hasIntervals()` is true, since you're setting the `isReadsToReferenceComparison` argument to true, and this checks the intervals.